client/my-sites/backup/rewind-flow/download-expired-plan.tsx

```diff
@@ -0,0 +1,281 @@
+import isGranularBackupDownloadRequested from 'calypso/state/selectors/is-granular-backup-download-requested';
+import Error from './error';
+import Loading from './loading';
+import ProgressBar from './progress-bar';
+import RewindConfigEditor from './rewind-config-editor';
+import RewindFlowNotice, { RewindFlowNoticeLevel } from './rewind-flow-notice';
+//import CheckYourEmail from './rewind-flow-notice/check-your-email';
+import { defaultRewindConfig, RewindConfig } from './types';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+interface Props {
+	backupDisplayDate: string;
+	rewindId: string;
+	siteId: number;
+	siteUrl: string;
+}
+
+// Future Work: Centralize typing
+interface BackupProgress {
+	downloadId: number;
+	progress?: number;
+	rewindId: string;
+	url?: string;
+	bytesFormatted: string;
+}
+
+const BackupDownloadFlowExpiredPlan: FunctionComponent< Props > = ( {
+	backupDisplayDate,
+	rewindId,
+	siteId,
+	siteUrl,
+} ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const [ userRequestedDownload, setUserRequestedDownload ] = useState( false );
+	const [ rewindConfig, setRewindConfig ] = useState< RewindConfig >( defaultRewindConfig );
+
+	const backupProgress = useSelector( ( state ) =>
+		getBackupProgress( state, siteId )
+	) as BackupProgress | null;
+
+	const requestedBackup = useSelector( ( state ) => getRequestedBackup( state, siteId ) );
+
+	const downloadId = backupProgress?.downloadId;
+	const downloadUrl = backupProgress?.url;
+	const downloadSize = backupProgress?.bytesFormatted;
+	const downloadProgress =
+		backupProgress && backupProgress.progress !== undefined && ! isNaN( backupProgress.progress )
+			? backupProgress.progress
+			: undefined;
+	const downloadRewindId = backupProgress?.rewindId;
+	const downloadInfoRequest = useSelector( ( state ) =>
+		getRequest( state, getRewindBackupProgress( siteId ) )
+	);
+
+	const isGranularRestore = useSelector( ( state ) =>
+		isGranularBackupDownloadRequested( state, siteId )
+	);
+
+	const requestDownload = useCallback(
+		() => dispatch( rewindBackup( siteId, rewindId, rewindConfig ) ),
+		[ dispatch, rewindConfig, rewindId, siteId ]
+	);
+	const trackedRequestDownload = useTrackCallback(
+		requestDownload,
+		'calypso_jetpack_backup_download_click'
+	);
+
+	const isDownloadInfoRequestComplete = downloadInfoRequest?.hasLoaded;
+	const isOtherDownloadInfo =
+		downloadRewindId !== rewindId || ( downloadId !== 0 && requestedBackup !== downloadId );
+	const isOtherDownloadInProgress = isOtherDownloadInfo && downloadProgress !== undefined;
+	const isDownloadURLNotReady = downloadUrl === undefined || downloadUrl === '';
+	const [ downloadFlowImageSrc, setDownloadFlowImageSrc ] = useState( JetpackBackupSVG );
+
+	useEffect( () => {
+		if ( ! isDownloadURLNotReady ) {
+			dispatch( recordTracksEvent( 'calypso_jetpack_backup_download_ready' ) );
+		}
+	}, [ isDownloadURLNotReady, dispatch ] );
```

Comment:
There is a bug that when a download completes, if you refresh the page it will always record `calypso_jetpack_backup_download_ready` Tracks event, because the `getBackupProgress` selector is fetching the recent download which the `url` is not empty nor `null`:

```
{
    "backupPoint": "2024-09-28T02:25:47+00:00",
    "downloadId": 485404,
    "progress": null,
    "rewindId": "1727490347",
    "startedAt": "2024-10-14T19:22:06+00:00",
    "downloadCount": 1,
    "validUntil": "2024-10-21T19:22:15+00:00",
    "bytesFormatted": "46 MB",
    "url": "https://public-api.wordpress.com/wpcom/v2/sites/(...)"
}
```

I think we can solve it by validating if the user requested the download, like this:

```suggestion
	useEffect( () => {
		if ( ! isDownloadURLNotReady && userRequestedDownload ) {
			dispatch( recordTracksEvent( 'calypso_jetpack_backup_download_ready' ) );
		}
	}, [ isDownloadURLNotReady, dispatch, userRequestedDownload ] );
```

Any thoughts?

Reply:
Agreed! I've gone ahead and added `userRequestedDownload` as a condition.